lib/logfmt/encoder.ex

```diff
@@ -1,3 +1,40 @@
+defprotocol Logfmt.ValueEncoder do
```

Comment:
A lot of these types already implement the `String.Chars` protocol, so I think it's fine to just return the value rather than going through the work of stringifying them here.

Reply:
Also, can we put the protocol in its own file and document it? I find it distracting if I'm trying to look at the encoder. Protocols in the Elixir source code [are their own files](https://github.com/elixir-lang/elixir/blob/1406d853e0e6515007696f871d0a9e2c023da6da/lib/elixir/lib/string/chars.ex#L3), as well.

Reply:
Regarding String.Chars

Enum.join will fail given a iolist, since it has mixed data and contain nested lists.

It works at the top-level

```elixir
Enum.join [:atom, " was an atom while ", 12, " was an integer"]
# => "atom was an atom while 12 was an integer"
```

But once you enter the world of iolists or list of lists...

```
iex(2)> Enum.join [[:atom, " was an atom while "], [12, " was an integer"]]
** (ArgumentError) cannot convert the given list to a string.

To be converted to a string, a list must contain only:

  * strings
  * integers representing Unicode codepoints
  * or a list containing one of these three elements

Please check the given list or call inspect/1 to get the list representation, got:

[:atom, " was an atom while "]

    (elixir) lib/list.ex:742: List.to_string/1
    (elixir) lib/enum.ex:1811: Enum."-join/2-lists^foldl/2-0-"/3
    (elixir) lib/enum.ex:1811: Enum.join/2
```